testsuite/MDAnalysisTests/analysis/test_msd.py

```diff
@@ -0,0 +1,192 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+from __future__ import division, absolute_import, print_function
+
+
+import MDAnalysis as mda
+from MDAnalysis.analysis.msd import EinsteinMSD as MSD
+
+from numpy.testing import (assert_array_less,
+                           assert_almost_equal, assert_equal)
+import numpy as np
+
+from MDAnalysisTests.datafiles import PSF, DCD, RANDOM_WALK, RANDOM_WALK_TOPO
+
+import pytest
+import tidynamics
+
+SELECTION = 'backbone and name CA and resid 1-10'
+NSTEP = 5000
+
+#universe
+@pytest.fixture(scope='module')
+def u():
+    return mda.Universe(PSF, DCD)
+
+@pytest.fixture(scope='module')
+def random_walk_u():
+    #100x100
+    return mda.Universe(RANDOM_WALK_TOPO, RANDOM_WALK)
+
+#non fft msd
+@pytest.fixture(scope='module')
+def msd(u):
+    m = MSD(u, SELECTION, msd_type='xyz', fft=False)
+    m.run()
+    return m
+
+#fft msd
+@pytest.fixture(scope='module')
+def msd_fft(u):
+    m = MSD(u, SELECTION, msd_type='xyz', fft=True)
+    m.run()
+    return m
+
+#all possible dimensions
+@pytest.fixture(scope='module')
+def dimension_list():
+    dimensions = ['xyz', 'xy', 'xz', 'yz', 'x', 'y', 'z']
+    return dimensions
+
+@pytest.fixture(scope='module')
+def step_traj(): # constant velocity
+    x = np.arange(NSTEP)
+    traj = np.vstack([x,x,x]).T
+    traj_reshape = traj.reshape([NSTEP,1,3])
+    u = mda.Universe.empty(1)
+    u.load_new(traj_reshape)
+    return u
+@pytest.fixture(scope='module')
+def step_traj_arr(): # constant velocity
+    x = np.arange(NSTEP)
+    traj = np.vstack([x,x,x]).T
+    return traj
+
+def random_walk_3d():
+    steps = -1 + 2*np.random.randint(0, 2, size=(NSTEP, 3))
```

Comment:
Should you seed this?

Reply:
The fixture can only be called once. Do you think its nessecary?

Reply:
Yeah it should get seeded so if someone has a bug with it one day I can reproduce it exactly.